packages/core/client.js

```diff
@@ -54,24 +55,45 @@ class BugsnagClient {
     this.BugsnagSession = BugsnagSession
   }
 
-  configure (opts = {}) {
-    this.config = config.mergeDefaults({ ...this.config, ...opts }, this.configSchema)
-    const validity = config.validate(this.config, this.configSchema)
+  setOptions (opts) {
+    this._opts = { ...this._opts, ...opts }
+  }
+
+  configure (partialSchema = config.schema) {
+    const conf = config.mergeDefaults(this._opts, partialSchema)
+    const validity = config.validate(conf, partialSchema)
+
     if (!validity.valid === true) throw new Error(generateConfigErrorMessage(validity.errors))
-    if (typeof this.config.beforeSend === 'function') this.config.beforeSend = [ this.config.beforeSend ]
-    if (this.config.appVersion !== null) this.app.version = this.config.appVersion
-    if (this.config.appType !== null) this.app.type = this.config.appType
-    if (this.config.metaData) this.metaData = this.config.metaData
-    if (this.config.user) this.user = this.config.user
-    if (this.config.logger) this.logger(this.config.logger)
+
+    // update and elevate some special options if they were passed in at this point
+    if (typeof conf.beforeSend === 'function') conf.beforeSend = [ conf.beforeSend ]
+    if (conf.appVersion) this.app.version = conf.appVersion
+    if (conf.appType) this.app.type = conf.appType
+    if (conf.metaData) this.metaData = conf.metaData
+    if (conf.user) this.user = conf.user
+    if (conf.logger) this.logger(conf.logger)
+
+    // merge with existing config
+    this.config = { ...this.config, ...conf }
+
     this._configured = true
+
     this._logger.debug(`Loaded!`)
```

Comment:
I realised that now `configure()` is called multiple times, this message needs to go somewhere else.